src/command/backup/backup.c

```diff
@@ -0,0 +1,1904 @@
+                    ioWriteFilterGroup(storageWriteIo(write)), gzipCompressNew((int)cfgOptionUInt(cfgOptCompressLevel), false));
+            }
+
+            // Add encryption filter if required
+            cipherBlockFilterGroupAdd(
+                filterGroup, cipherType(cfgOptionStr(cfgOptRepoCipherType)), cipherModeEncrypt, manifestCipherSubPass(manifest));
+
+            // Add size filter last to calculate repo size
+            ioFilterGroupAdd(filterGroup, ioSizeNew());
+
+            // Write file
+            storagePutP(write, BUFSTR(content));
+
+            // Use base path to set ownership and mode
+            const ManifestPath *basePath = manifestPathFind(manifest, MANIFEST_TARGET_PGDATA_STR);
+
+            // Add to manifest
+            ManifestFile file =
+            {
+                .name = manifestName,
+                .primary = true,
+                .mode = basePath->mode & (S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH),
+                .user = basePath->user,
+                .group = basePath->group,
+                .size = strSize(content),
+                .sizeRepo = varUInt64Force(ioFilterGroupResult(filterGroup, SIZE_FILTER_TYPE_STR)),
+                .timestamp = timestamp,
+            };
+
+            memcpy(
+                file.checksumSha1, strPtr(varStr(ioFilterGroupResult(filterGroup, CRYPTO_HASH_FILTER_TYPE_STR))),
+                HASH_TYPE_SHA1_SIZE_HEX + 1);
+
+            manifestFileAdd(manifest, &file);
+
+            LOG_DETAIL_FMT("wrote '%s' file returned from pg_stop_backup()", strPtr(name));
+        }
+        MEM_CONTEXT_TEMP_END();
+    }
+
+    FUNCTION_LOG_RETURN_VOID();
+}
+
+/*--------------------------------------------------------------------------------------------------------------------------------*/
+typedef struct BackupStopResult
+{
+    String *lsn;
+    String *walSegmentName;
+    time_t timestamp;
+} BackupStopResult;
+
+#define FUNCTION_LOG_BACKUP_STOP_RESULT_TYPE                                                                                       \
+    BackupStopResult
+#define FUNCTION_LOG_BACKUP_STOP_RESULT_FORMAT(value, buffer, bufferSize)                                                          \
+    objToLog(&value, "BackupStopResult", buffer, bufferSize)
+
+static BackupStopResult
+backupStop(BackupData *backupData, Manifest *manifest)
+{
+    FUNCTION_LOG_BEGIN(logLevelDebug);
+        FUNCTION_LOG_PARAM(BACKUP_DATA, backupData);
+        FUNCTION_LOG_PARAM(MANIFEST, manifest);
+    FUNCTION_LOG_END();
+
+    BackupStopResult result = {.lsn = NULL};
+
+    if (cfgOptionBool(cfgOptOnline))
+    {
+        MEM_CONTEXT_TEMP_BEGIN()
+        {
+            // Stop the backup
+            LOG_INFO_FMT(
+                "execute %sexclusive pg_stop_backup() and wait for all WAL segments to archive",
+                backupData->version >= PG_VERSION_96 ? "non-" : "");
+
+            DbBackupStopResult dbBackupStopResult = dbBackupStop(backupData->dbPrimary);
```

Comment:
The dbBackupStop query used to have a timestamp in it like the backupStart query but in the new db.c code it does not. Form consistency maybe we should have it for both start and stop or for neither.

Reply:
Hmmm, dbBackupStart() doesn't have a timestamp.  I'm not sure what you mean.

Reply:
String *result = strNewFmt(
        "select lsn::text as lsn,\n"
        "       pg_catalog.pg_%sfile_name(lsn)::text as wal_segment_name\n"
        "  from pg_catalog.pg_start_backup('" PROJECT_NAME " backup started at ' **|| current_timestamp**",
        strPtr(pgWalName(pgVersion)));